tests/pytests/unit/states/test_jboss7.py

```diff
@@ -0,0 +1,756 @@
+        else:
+            return {"success": True, "result": {"value": "DEV"}}
+
+    def update_func(jboss_config, binding_name, value, profile):
+        binding_status["updated"] = True
+        return {"success": True}
+
+    read_mock = MagicMock(side_effect=read_func)
+    create_mock = MagicMock()
+    update_mock = MagicMock(side_effect=update_func)
+
+    with patch.dict(
+        jboss7.__salt__,
+        {
+            "jboss7.read_simple_binding": read_mock,
+            "jboss7.create_simple_binding": create_mock,
+            "jboss7.update_simple_binding": update_mock,
+        },
+    ):
+        # when
+        result = jboss7.bindings_exist(
+            name="bindings", jboss_config={}, bindings={"env": "DEV2"}, profile=None
+        )
+
+        # then
+        update_mock.assert_called_with(
+            jboss_config={}, binding_name="env", value="DEV2", profile=None
+        )
+        assert create_mock.call_count == 0
+        assert result["changes"] == {"changed": "env:DEV->DEV2\n"}
+        assert result["comment"] == "Bindings changed."
+
+
+def test_should_not_update_bindings_if_same():
+    # given
+    read_mock = MagicMock(return_value={"success": True, "result": {"value": "DEV2"}})
+    create_mock = MagicMock()
+    update_mock = MagicMock()
+
+    with patch.dict(
+        jboss7.__salt__,
+        {
+            "jboss7.read_simple_binding": read_mock,
+            "jboss7.create_simple_binding": create_mock,
+            "jboss7.update_simple_binding": update_mock,
+        },
+    ):
+        # when
+        result = jboss7.bindings_exist(
+            name="bindings", jboss_config={}, bindings={"env": "DEV2"}
+        )
+
+        # then
+        assert create_mock.call_count == 0
+        assert update_mock.call_count == 0
+        assert result["changes"] == {}
+        assert result["comment"] == "Bindings not changed."
+
+
+def test_should_raise_exception_if_cannot_create_binding():
+    def read_func(jboss_config, binding_name, profile):
+        return {"success": False, "err_code": "JBAS014807"}
+
+    def create_func(jboss_config, binding_name, value, profile):
+        return {"success": False, "failure-description": "Incorrect binding name."}
+
+    read_mock = MagicMock(side_effect=read_func)
+    create_mock = MagicMock(side_effect=create_func)
+    update_mock = MagicMock()
+
+    with patch.dict(
+        jboss7.__salt__,
+        {
+            "jboss7.read_simple_binding": read_mock,
+            "jboss7.create_simple_binding": create_mock,
+            "jboss7.update_simple_binding": update_mock,
+        },
+    ):
+        # when
+        try:
+            jboss7.bindings_exist(
+                name="bindings",
+                jboss_config={},
+                bindings={"env": "DEV2"},
+                profile=None,
+            )
+            pytest.fail("An exception should be thrown")
+        except CommandExecutionError as e:
+            assert str(e) == "Incorrect binding name."
```

Comment:
```suggestion
        with pytest.raises(CommandExecutionError) as exc:
            jboss7.bindings_exist(
                name="bindings",
                jboss_config={},
                bindings={"env": "DEV2"},
                profile=None,
            )
        assert str(exc.value) == "Incorrect binding name."
```
This is how we should test code that should raise exceptions